### PR TITLE
Add additional non-sparse tests

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

File: test/replicate.js [hunk 1]
```diff
@@ -737,4 +737,23 @@ test('download available blocks on non-sparse update', async function (t) {
...
+  await a.append(['a', 'b', 'c', 'd', 'e'])
+  await b.update()
+
+  const s = b.snapshot()
+
+  await a.append(['f', 'g', 'h', 'i', 'j'])
+  await s.update()
+
+  t.is(s.contiguousLength, 5)
+  t.is(s.length, s.contiguousLength)
```
Comment (mafintosh) [inline test/replicate.js:758]: @kasperisager can you add one with partial replication also? where length will be !== contigLength

Comment (kasperisager) [reply test/replicate.js:758]: I'm not entirely sure how I'd accomplish that when the non-sparse session continuously replicates.

Comment (mafintosh) [reply test/replicate.js:758]: @kasperisager have a peer inbetween that is sparse, where you only replicate specific blocks over (i think i added a similar test)

Comment (kasperisager) [reply test/replicate.js:758]: Ah, makes sense!

Comment (mafintosh) [reply test/replicate.js:758]: @kasperisager i don't think we need any contigLength guarantees around snapshots etc etc, it's an internal detail like .fork

**Pull request merged**